web/src/app/services/IntegrationKeyList.js

```diff
@@ -53,134 +55,112 @@ const sortItems = (a, b) => {
+      subText: (
+        <IntegrationKeyDetails
+          key={key.id}
+          href={key.href}
+          label={typeLabels[key.type]}
+          type={key.type}
+          classes={props.classes}
         />
-        {this.state.create && (
-          <IntegrationKeyCreateDialog
-            serviceID={this.props.serviceID}
-            onClose={() => this.setState({ create: false })}
-          />
-        )}
-        {this.state.delete && (
-          <IntegrationKeyDeleteDialog
-            integrationKeyID={this.state.delete}
-            onClose={() => this.setState({ delete: null })}
-          />
-        )}
-      </React.Fragment>
-    )
-  }
-
-  renderQuery() {
-    return (
-      <Query
-        query={query}
-        variables={{ serviceID: this.props.serviceID }}
-        render={({ data }) => this.renderList(data.service.integrationKeys)}
+      ),
+      secondaryAction: (
+        <IconButton onClick={() => setDeleteDialog(key.id)}>
+          <Trash />
+        </IconButton>
+      ),
+    }))
+
+  return (
+    <React.Fragment>
+      <Grid item xs={12} className={classes.spacing}>
+        <Card>
+          <CardContent>
+            <FlatList
+              data-cy='int-keys'
+              headerNote={
+                <React.Fragment>
+                  API Documentation is available{' '}
+                  <AppLink to='/docs'>here</AppLink>.
+                </React.Fragment>
+              }
+              emptyMessage='No integration keys exist for this service.'
+              items={items}
+            />
+          </CardContent>
+        </Card>
+      </Grid>
+      <CreateFAB
+        onClick={() => setCreate(true)}
+        title='Create Integration Key'
       />
-    )
-  }
-
-  renderList(keys) {
-    const typeLabels = {
-      generic: 'Generic API Key',
-      grafana: 'Grafana Webhook URL',
-      site24x7: 'Site24x7 Webhook URL',
-      email: 'Email Address',
-    }
+      {create && (
+        <IntegrationKeyCreateDialog
+          serviceID={props.serviceID}
+          onClose={() => setCreate(false)}
+        />
+      )}
+      {deleteDialog && (
+        <IntegrationKeyDeleteDialog
+          integrationKeyID={deleteDialog}
+          onClose={() => setDeleteDialog(null)}
+        />
+      )}
+    </React.Fragment>
+  )
+}
 
-    const items = (keys || [])
-      .slice()
-      .sort(sortItems)
-      .map((key) => ({
-        title: key.name,
-        subText: (
-          <IntegrationKeyDetails
-            key={key.id}
-            href={key.href}
-            label={typeLabels[key.type]}
-            type={key.type}
-            classes={this.props.classes}
-          />
-        ),
-        secondaryAction: (
-          <IconButton onClick={() => this.setState({ delete: key.id })}>
-            <Trash />
-          </IconButton>
-        ),
-      }))
+IntegrationKeyDetails.propTypes = {
+  href: p.string.isRequired,
+  label: p.string.isRequired,
+  type: p.string.isRequired,
+}
```

Comment:
Move this up with `IntegrationKeyDetails`

Reply:
I moved it right underneath `IntegrationKeyDetails`